central/group/datastore/filter/filter.go

```diff
@@ -0,0 +1,50 @@
+package filter
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/globaldb"
+	store2 "github.com/stackrox/rox/central/group/datastore/store"
+	"github.com/stackrox/rox/central/group/datastore/store/bolt"
+	"github.com/stackrox/rox/central/group/datastore/store/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+func GetFiltered(ctx context.Context, filter func(*storage.Group) bool, store store2.Store) ([]*storage.Group, error) {
+	var groups []*storage.Group
+	walkFn := func() error {
+		groups = groups[:0]
+		return store.Walk(ctx, func(g *storage.Group) error {
+			if filter == nil || filter(g) {
+				groups = append(groups, g)
+			}
+			return nil
+		})
+	}
+	if err := pgutils.RetryIfPostgres(walkFn); err != nil {
+		return nil, err
+	}
+	return groups, nil
+}
+
+var (
+	store store2.Store
+	once  sync.Once
+)
+
+func initialize() {
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		store = postgres.New(globaldb.GetPostgres())
+	} else {
+		store = bolt.New(globaldb.GetGlobalDB())
+	}
+}
+
+// GroupStoreSingleton returns the singleton providing access to the roles store.
+func GroupStoreSingleton() store2.Store {
+	once.Do(initialize)
+	return store
+}
```

Comment:
Instead of this one, we could also have `GetFiltered` call this singleton internally and not expose the `store.Store` interface at all, right?

In this case, you could just inject within role / auth provider datastore a function type and mock it as such, instead of injecting the `store.Store` interface which provides way more access than just the singular function to clients.